utils/scarb-ui/src/counter.rs

```diff
@@ -0,0 +1,27 @@
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+#[derive(Default)]
+pub struct DiagnosticsCounter {
+    pub errors: AtomicUsize,
+    pub warnings: AtomicUsize,
+}
+
+impl DiagnosticsCounter {
+    pub fn error(&self) {
+        let _ = self.errors.fetch_add(1, Ordering::Release);
+    }
+    pub fn warning(&self) {
+        let _ = self.warnings.fetch_add(1, Ordering::Release);
+    }
+    pub fn finish(&self) -> DiagnosticsCount {
+        DiagnosticsCount {
+            errors_count: self.errors.load(Ordering::Acquire),
+            warnings_count: self.warnings.load(Ordering::Acquire),
+        }
+    }
+}
+
+pub struct DiagnosticsCount {
+    pub errors_count: usize,
+    pub warnings_count: usize,
```

Comment:
nit. field names duplicated struct name


```suggestion
    pub errors: usize,
    pub warnings: usize,
```